float64 pd
float64 thetad
